### PR TITLE
chore(examples): Modernize JS examples

### PR DESCRIPTION
**Pull request by marcoscaceres**
Make the examples a little more 2015... for the millennials.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -912,16 +912,19 @@ spec: webidl
...
-      navigator.permissions.query({name:'geolocation'}).then(function(result) {
-        if (result.state == 'granted') {
-          showLocalNewsWithGeolocation();
-        } else if (result.state == 'prompt') {
-          showButtonToEnableLocalNews();
-        }
-        // Don't do anything if the permission was denied.
-      });
-    &lt;/script&gt;
+navigator.permissions.query({ name: "geolocation" }).then(({ state }) => {
```
Comment (jyasskin) [inline index.bs]: With bikeshed, you can leave things inside `<pre>` indented, and it'll figure out the indentation level.

File: index.bs [hunk 2]
```diff
@@ -953,33 +954,42 @@ spec: webidl
...
+    console.log("Geolocation permission state is:", geoState);
+    console.log("Notifications permission state is:", notifState);
+  });
   </pre>
   </div>
   <div class="example" id="example-deviceId">
   <p>
-    This example is checking whether the page has the {{"camera"}}
-    permission for the first device in the list. Error checking omitted.
+    This example is checking the permission of the the first camera.
```
Comment (jyasskin) [inline index.bs]: s/is checking/checks/? 

Can you say what "the permission of" means. Is it permission to record video from, permission to see the metadata of, or what?

You have a "the the".

File: index.bs [hunk 3]
```diff
@@ -953,33 +954,42 @@ spec: webidl
...
-    &lt;/script&gt;
+navigator.mediaDevices
+  .enumerateDevices()
+  // filter out just video inputs
+  .then(
+    devices => devices.filter(({ kind }) => kind === "videoinput")
+  )
+  // get the deviceId of first camera, create query object
+  .then(
+    ([{ deviceId }]) => ({ name: "camera", deviceId })
```
Comment (jyasskin) [inline index.bs]: It seems unnecessarily verbose to create the query object and call `query()` in two separate `.then()` steps. Combine them?

File: index.bs [hunk 4]
```diff
@@ -953,33 +954,42 @@ spec: webidl
...
+  // perform query
+  .then(
+    queryObj => navigator.permissions.query(queryObj)
+  )
+  // log the state
+  .then(
+    ({ state }) => console.log("Permission state of first camera is", state)
+  )
+  .catch(
+    (err) => console.error(err.stack)
```
Comment (jyasskin) [inline index.bs]: You don't need the ()s around `err` here. Do we want to be consistent about being completely minimal, or do we not care?

**Comment by jyasskin** [comment]
Otherwise, looks good to me.

**Comment by jyasskin** [comment]
Oh, except, what's "chore(...):" in the commit titles? If it's just personal bookkeeping, we should probably keep it out of the commit log.

File: index.bs (hunk 1 above)
Comment (marcoscaceres) [inline index.bs]: Will re-indent.

File: index.bs (hunk 3 above)
Comment (marcoscaceres) [inline index.bs]: Yep, good suggestion. Can do basically, filter, and then map to query object, and then map to promise there... and promise.all() for the cameras.

File: index.bs (hunk 4 above)
Comment (marcoscaceres) [inline index.bs]: Let's go with minimal... no convention has really emerged yet, AFAIK, regarding these shorthands... but omitting optional bits sounds good unless they are actually needed.

**Comment by marcoscaceres** [comment]
> Oh, except, what's "chore(...):" in the commit titles? If it's just personal bookkeeping, we should probably keep it out of the commit log.

It's a convention for better tracking of changes, and for tracking the context of where changes are made. It's loosely based on [Karma's commit conventions](http://karma-runner.github.io/0.10/dev/git-commit-msg.html). I would encourage us to adopt it because it makes it easier to understand what has changed and why.

**Comment by marcoscaceres** [comment]
... refactor(examples) is probably a more accurate description of what changed here.

**Comment by marcoscaceres** [comment]
Addressed the feedback... please merge if all looks ok now.

**Pull request merged**